silabs-pti/src/main/java/com/silabs/pti/Main.java

```diff
@@ -200,7 +212,15 @@ private <T> int runCapture(final IDebugChannelExportFormat<T> format,
 
           // Admin connection / configure Time Server
           if (!cli.testMode() && cli.discreteNodeCapture() == false && cli.hostnames().length > 1) {
-            final IConnection admin = adapterConnector.createConnection(ip, AdapterPort.ADMIN.defaultPort(), cli);
+            hostName = ip;
+            aPort = AdapterPort.ADMIN.defaultPort();
```

Comment:
This is essentially same block of code than above at line 198.

Reply:
I'd prefer to do this in a following way:

1.) Create a new object: "AddressAndPort", and make it contains "String address" and "int port".
2.) Create a new method "parseAddressAndPort(String)" that returns it, and it NEVER returns null. If there is no ':' in the string, it returns the object that simply contains address and default port. If it has ':' then parse the port out of it.
3.) This block of code then does not have to do "if (addrAndPort != null)". It simply does `parseAddressAndPort()` and uses whatever comes back.

This way, there is less duplication.